File: food-backend/src/main/java/com/licenta/food_backend/entity/Meal.java
package com.licenta.food_backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;
import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "meals")
public class Meal {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private String foodName;

    @Column(nullable = false)
    private Integer caloriesPer100g;

    private Double weightInGrams;

    private Double totalCalories;

    private LocalDateTime consumedAt;
    
    @PrePersist
    protected void onCreate() {
        this.consumedAt = LocalDateTime.now();
    }

    @Column(name = "meal_date")
    private LocalDate date;

    @Column(nullable = true)
    private Double protein;

    @Column(nullable = true)
    private Double carbs;

    @Column(nullable = true)
    private Double fats;
}

File: food-frontend-angular/src/app/core/types/meal.ts
export interface Meal {
  id: number;
  foodName: string;
  caloriesPer100g: number;
  weightInGrams: number | null;
  totalCalories: number | null;
  consumedAt: string | null;
  date: string;
  protein: number | null;
  carbs: number | null;
  fats: number | null;
}

export interface DailyTotals {
  calories: number;
  protein: number;
  carbs: number;
  fats: number;
}

export interface MealSavePayload {
  foodName: string;
  caloriesPer100g: number;
  weightInGrams: number;
  protein: number;
  carbs: number;
  fats: number;
  date?: string;
}

export interface AiPrediction {
  rank: number;
  food: string;
  calories_per_100g: number;
  protein_per_100g: number;
  carbs_per_100g: number;
  fats_per_100g: number;
  confidence_percentage: number;
}
